components/Dialogs/src/MDCAlertController.h

```diff
@@ -132,15 +116,66 @@
 @property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
     BOOL mdc_adjustsFontForContentSizeCategory;
 
+/**
+ The actions that the user can take in response to the alert.
+
+ The order of the actions in the array matches the order in which they were added to the alert.
+ */
+@property(nonatomic, nonnull, readonly) NSArray<MDCAlertAction *> *actions;
+
+/**
+ Adds an action to the alert dialog.
+
+ Actions are the possible reactions of the user to the presented alert. Actions are added as a
+ button at the bottom of the alert. Affirmative actions should be added before dismissive actions.
+ Action buttons will be laid out from right to left if possible or top to bottom depending on space.
+
+ Material spec recommends alerts should not have more than two actions.
+
+ @param action Will be added to the end of MDCAlertController.actions.
+ */
+- (void)addAction:(nonnull MDCAlertAction *)action;
+
+/**
+ Returns an MDCButton associated with the given action. This method might create the button if
+ it no associated button exist for the action yet. Buttons returned by thismethod may not (yet)
+ be attached to the view hierarchy at the time the method is called.
+
+ This method is commonly used by themers to style the button associated with the action.
+
+ @param action The action with which the button is associated. Must be an existing action that
+               had been previously added through addAction: to the alert.
+ @return The button associated with the action, or nil if the action doesn't exist (the action
+          must first be added to the alert).
+ */
+- (nullable MDCButton *)buttonForAction:(nonnull MDCAlertAction *)action;
+
 /** MDCAlertController handles its own transitioning delegate. */
 - (void)setTransitioningDelegate:
-        (_Nullable id<UIViewControllerTransitioningDelegate>)transitioningDelegate NS_UNAVAILABLE;
+    (_Nullable id<UIViewControllerTransitioningDelegate>)transitioningDelegate NS_UNAVAILABLE;
 
 /** MDCAlertController.modalPresentationStyle is always UIModalPresentationCustom. */
 - (void)setModalPresentationStyle:(UIModalPresentationStyle)modalPresentationStyle NS_UNAVAILABLE;
 
+/**
+ Set this flag to true to enable emphasis-based theming of dialog buttons. Setting the flag to
+ true ignores buttonFont, buttonTitleColor & buttonInkColor, and replaces these with button
+ theming applied by a dialog themer.
+
+ Default value: NO
+
+ b/117717380: will be deprecated
+ */
+@property(nonatomic, assign) BOOL mdc_enableEmphasisThemingOfActionButtons;
```

Comment:
The `mdc_` prefix is surprising here - can we remove it? We typically only use this prefix in categories of classes we do not own.

Reply:
I'm also concerned about adding APIs that imply the existence of a theming system to our components. Can we move this logic to the themer APIs instead by introducing new APIs that handle the new, desired behavior?

Reply:
+1 Please remove this property. Emphasis should always be used.

Reply:
This should not be needed, since by default legacy will be "low emphasis" which equates to Text Buttons. Since Text Buttons are the "legacy" (or current) style for buttons in an Alert, we should expect no change to any clients that do not adopt the new styling.  

@randallli  Can you please explain why you believe a separate flag (in addition to the `emphasis` argument) is required?

Reply:
This flag is now removed.